presto-main/src/test/java/com/facebook/presto/server/TestServer.java

```diff
@@ -229,4 +256,27 @@ public URI uriFor(String path)
     {
         return HttpUriBuilder.uriBuilderFrom(server.getBaseUrl()).replacePath(path).build();
     }
+    private QueryInfo runQuery(String query)
+            throws Exception
+    {
+        Request request = preparePost()
+                .setUri(uriFor("/v1/statement"))
+                .setHeader(PRESTO_USER, "user")
+                .setHeader(PRESTO_CATALOG, "catalog")
+                .setHeader(PRESTO_SCHEMA, "schema")
+                .setBodyGenerator(createStaticBodyGenerator(query, UTF_8))
+                .build();
+
+        QueryResults queryResults = client.execute(request, createJsonResponseHandler(jsonCodec(QueryResults.class)));
+
+        while (queryResults.getNextUri() != null) {
+            queryResults = client.execute(prepareGet().setUri(queryResults.getNextUri()).build(), createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        }
+
+        QueryInfo queryInfo;
+        do {
+            queryInfo = server.getQueryManager().getQueryInfo(new QueryId(queryResults.getId()));
+        } while (!queryInfo.getState().isDone());
```

Comment:
We shouldn't need a loop here. After the above loop exits, the query will already be done.